### DISCUSSION TITLE
Network Storage

### DISCUSSION THREAD

**Discussion by DivanX10** [doitandbedone/ispyagentdvr-docker#469] [General]
AgentDVR v.3.5.4.0

Tell me, how do I correctly specify the path to the Home Assistant network folder? In the configuration of the samba Home Assistant addon in the settings, I specified a username and password and there are no examples in the [instructions](https://www.ispyconnect.com/userguide-agent-network-storage.aspx) of how to specify the address to connect to the seed folder

For example, to connect to samba, you can use this option, but not in the case of AgentDVR
smb://user_name:password@server_name

```
workgroup: WORKGROUP
username: user
password: pass
allow_hosts:
  - 10.0.0.0/8
  - 172.16.0.0/12
  - 192.168.0.0/16
  - fe80::/10
veto_files:
  - ._*
  - .DS_Store
  - Thumbs.db
  - icon?
  - .Trashes
compatibility_mode: false
```

I tried different options and the files are not saved to this network folder
![image](https://user-images.githubusercontent.com/64090632/136120141-189c6236-9b6c-4e52-a23c-4ea6c519e8ef.png)
![image](https://user-images.githubusercontent.com/64090632/136120402-b2f47e7f-7965-45ea-b16e-32061df00290.png)

**Comment by doitandbedone** [comment]
Hello, you have to remember that docker containers run on a sandbox and only see what you allow them to see. In this case, the container runs on it's own network and is only allowed to talk to other services via the ports allowed, which you have specified on the run command.

With that said, for this to work, you would have to include some sort of plug in or communication channel for the container to reach your samba share. I did a quick search and some things popped up. 

Check this thread:
https://stackoverflow.com/questions/27989751/mount-smb-cifs-share-within-a-docker-container

If everything else fails, you can try using agent outside of docker, but to be honest I'm not sure if agent supports it out of the box. 

@ispysoftware can you elaborate on support for SMB shares?

**Comment by ispysoftware** [comment]
I don't know much about connecting to samba shares from within docker
containers but according to microsoft docs:
https://social.msdn.microsoft.com/Forums/vstudio/en-US/03bf0cf2-be80-43a4-870c-83727dee1c63/access-a-samba-share-via-c?forum=csharpgeneral
it should just be a case of using the path like:
\\SERVER\media\AgentDVR\
and making sure the container has permission to write to that location

maybe it's the addition of the / at the end of the path that's breaking
something.

**Comment by DivanX10** [comment]
I then mounted the docker disk to the home assistant, it remains to be done as in the instructions.  Only I have already turned off the PC and go to bed.  I'll finish it in the afternoon.  I just wish the agent DVR could place files on network drives

https://community.home-assistant.io/t/mount-remote-smb-share-on-hassio/116734/28?u=divanx10

**Comment by ispysoftware** [comment]
it can if you use windows, you can add network locations.

it looks like you have to use a library to write to samba shares from
linux, something like:

https://github.com/ume05rw/EzSmb

unfortunately adding support for that is a *lot *of work

**Comment by DivanX10** [comment]
I went the other way. Implemented by means of Synology. 
Created the HAOS folder at /agent/media with the mkdir HAOS command
![image](https://user-images.githubusercontent.com/64090632/136193508-e6d6b01e-f101-4dd1-aa59-680bd8e495db.png)

In Synology, in the Docker network folder, I created the AgentDVR folder where I mounted the Media network folder from the home assistant
![image](https://user-images.githubusercontent.com/64090632/136193719-7d6de672-94ac-4a65-b4ee-a59f4b2fa54a.png)
![image](https://user-images.githubusercontent.com/64090632/136192699-3fbcb7d8-69bb-4013-bbe9-f48ae7b26b01.png)

In the container settings, in the volumes section, I specified where to put the files
![image](https://user-images.githubusercontent.com/64090632/136192357-995fa99c-61d6-4795-9b93-ea6dd9f32390.png). 

In the storage settings, I specified the path to the HAOS folder
![image](https://user-images.githubusercontent.com/64090632/136193084-58989452-124e-4d32-845e-6b7c9f126ab7.png)

The Agent DVR folder will appear in the Home Assistant
![image](https://user-images.githubusercontent.com/64090632/136194401-c0c1c9b7-29f7-410b-9b9a-e500da38da60.png)

**Comment by DivanX10** [comment]
AgentDVR has an FTP setup. Is it possible to do the same, but for samba? Create a separate setting where there will be a window where we specify the path to the network folder where we need to put the video and login\password

![image](https://user-images.githubusercontent.com/64090632/136194873-216df3bf-9f42-4132-bc56-d73915af7560.png)
![image](https://user-images.githubusercontent.com/64090632/136195103-1295a978-16e3-439b-819e-4010aa3be6ea.png)

**Comment by ispysoftware** [comment]
There must be some other software that does that like copies content from a
local directory to a samba share?

**Comment by ispysoftware** [comment]
You could just put an ftp server on the target computer. Loads of free ones
available.

**Comment by DivanX10** [comment]
I have FTP on Synology, but we are talking about a home assistant and there I use the [samba addon](https://github.com/home-assistant/addons/tree/master/samba). This is the most convenient way to access network folders. Therefore, it would be more logical to add a samba setting to AgentDVR. I found a way for Synology, but not everyone has Synology and raises AgentDVR on different machines. I also added the [Folder Watcher integration](https://www.home-assistant.io/integrations/folder_watcher), which tracks files in the /media/agent DVD/video/VPIRD folder/ 
Thanks to this, I can send videos in a telegram

![image](https://user-images.githubusercontent.com/64090632/136207658-941dd612-d0ef-4cf4-b6e2-5b26153969e3.png)

**Comment by DivanX10** [comment]
I made a [manual](https://community.home-assistant.io/t/agentdvr-nas-synology-telegram/344520) how it can be implemented

**Comment by DivanX10** [comment]
If you can add samba to Agent DVR, then don't close this post, if you still can't do it, then it's okay. then close this question.

**Comment by doitandbedone** [comment]
Again I would recommend on looking for plugins on how to expose samba
shares as volumes for docker. This would be the fatest route for you.

**Comment by ispysoftware** [comment]
That's cool - it seems you know more about home assistant than I do - maybe
you could write the addon? it looks quite straightforward..

https://developers.home-assistant.io/docs/add-ons/tutorial/

**Comment by DivanX10** [comment]
I'm just an experienced user).  I can't write programs or add-ons.  Although, if you think about it, then you need a plug-in for the Agent DVR itself so that it can mount the Home Assistant network drive and will put the video into the Media folder
